services/director-v2/src/simcore_service_director_v2/api/dependencies/dynamic_sidecar.py

```diff
@@ -0,0 +1,17 @@
+from fastapi import Request
+
+from ...core.settings import DynamicSidecarSettings
+from ...modules.dynamic_sidecar.api_client import DynamicSidecarClient
+from ...modules.dynamic_sidecar.scheduler import DynamicSidecarsScheduler
+
+
+def get_dynamic_sidecar_client(request: Request) -> DynamicSidecarClient:
+    return request.app.state.dynamic_sidecar_api_client
```

Comment:
So the state concept for the dynamic-sidecar is encapsulated inside the `SchedulerData` object. It is currently stored in the `DynamicSidecarsScheduler` (already exposed here).
I see now utility into having it available here, since it is generally used internally by the scheduler.